core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -151,19 +143,28 @@ public void testUnsupportedKeyValueCollectionOperations() {
       List newObjCol = new ArrayList();
       newObjCol.add(newObj);
       for (Collection col : collections) {
-         try {
-            col.add(newObj);
-            assert false : "Should have thrown a UnsupportedOperationException";
-         } catch (UnsupportedOperationException uoe) {
-         } catch (ClassCastException e) {
-            // Ignore class cast in expired filtered set because
-            // you cannot really add an Object type instance.
-         }
-         try {
-            col.addAll(newObjCol);
-            assert false : "Should have thrown a UnsupportedOperationException";
-         } catch (UnsupportedOperationException uoe) {
-         }
+         col.add(newObj);
```

Comment:
I think that you want to test adding to both keys and values - this will throw exception with keys and the values won't be tested.
IMO having `Collection[] { keys, values }` is a strange pattern; I would rather use `keys.add(newObj);` and `values.add(newObj)` separately.
Note that instead of try-catch or `@Test(expectedExceptions = ...)` you can also use `Exceptions.expectException(...)`.